pkg/storage/postgres/schema.go

```diff
@@ -56,28 +56,46 @@ CREATE TABLE task_run (
 	id serial PRIMARY KEY,
 	vcs_id integer REFERENCES vcs(id),
 	task_id integer NOT NULL REFERENCES task (id) ON DELETE CASCADE,
+	total_input_digest text NOT NULL,
 	start_timestamp timestamp with time zone NOT NULL,
 	stop_timestamp timestamp with time zone NOT NULL,
 	result text NOT NULL,
 	CONSTRAINT result_check CHECK (result in ('success', 'failure'))
 );
+CREATE INDEX idx_task_run_total_input_digest ON task_run(total_input_digest);
 
-CREATE TABLE input (
+CREATE TABLE input_file (
 	id serial PRIMARY KEY,
-	uri text NOT NULL,
+	path text NOT NULL,
+	digest text NOT NULL,
+	CONSTRAINT input_file_path_digest_uniq UNIQUE (path, digest)
+);
+CREATE INDEX idx_input_file_path ON input_file(path);
```

Comment:
Do we need this index?

Reply:
`baur ls runs --input-string=PATH` searches in the table for a given path, the index should speed up those look ups.

Same situation  for input string + `baur ls runs --input-string=string:hello` but we cannot have an index on the string column.
The command+query can be changed  for input-strings instead to create the digest for the string and query query for the indexed digest in the table instead. The change can be done if that will ever become an issue.